src/protocol/sort/bit_permutation.rs

```diff
@@ -53,13 +55,13 @@ pub async fn bit_permutation<F: Field>(
         .map(|(i, (ctx, (x, sum)))| async move { ctx.multiply(RecordId::from(i), &x, &sum).await });
     let mut mult_output = try_join_all(async_multiply).await?;
 
-    assert_eq!(mult_output.len(), input.len() * 2);
+    debug_assert!(mult_output.len() == input.len() * 2);
     // Generate permutation location
     let len = mult_output.len() / 2;
     for i in 0..len {
         // we are subtracting "1" from the result since this protocol returns 1-index permutation whereas all other
         // protocols expect 0-indexed permutation
-        let less_one = &mult_output[i + len] - &share_of_one;
+        let less_one = mult_output[i + len].clone() - &share_of_one;
```

Comment:
Why did you change this to clone()?

Reply:
We are now performing S + &S vs &S + &S since addition of two &S isnt supported as yet.